Bugzilla 275 - XIP issues
Applied TLSbo85925_fix_XIP_booting_with_NOR_driver_enabled.patch

cvs diff -puN -r1.2 -r1.3 arch/arm/Kconfig
Index: arch/arm/Kconfig
===================================================================
RCS file: arch/arm/Kconfig,v
retrieving revision 1.2
retrieving revision 1.3
diff -p -u -r1.2 -r1.3
--- linux-2.6.19.2-mx/arch/arm/Kconfig	15 Jan 2007 05:44:47 -0000	1.2
+++ linux-2.6.19.2-mx/arch/arm/Kconfig	16 Jan 2007 23:56:03 -0000	1.3
@@ -260,6 +260,7 @@ config ARCH_L7200
 config ARCH_MXC
 	bool "Freescale MXC/iMX-based"
 	select ISA
+	select ARCH_MTD_XIP
 	help
 	  Support for Freescale MXC/iMX-based family of processors
 
cvs diff -puN -r1.0 -r1.1 include/asm-arm/arch-mxc/mtd-xip.h
Index: include/asm-arm/arch-mxc/mtd-xip.h
===================================================================
RCS file: include/asm-arm/arch-mxc/mtd-xip.h
diff -N include/asm-arm/arch-mxc/mtd-xip.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ linux-2.6.19.2-mx/include/asm-arm/arch-mxc/mtd-xip.h	16 Jan 2007 23:56:04 -0000	1.1
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2007 Freescale Semiconductor, Inc. All Rights Reserved.
+ */
+
+/*
+ * MTD primitives for XIP support. Architecture specific functions
+ *
+ * Do not include this file directly. It's included from linux/mtd/xip.h
+ */
+
+/*
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#ifndef __ARCH_MXC_MTD_XIP_H__
+#define __ARCH_MXC_MTD_XIP_H__
+
+#include <asm/hardware.h>
+#define xip_irqpending()        \
+        (~(__raw_readl(AVIC_NIVECSR) & __raw_readl(AVIC_FIVECSR)))
+#define xip_currtime()          get_cycles()
+#define xip_elapsed_since(x)    \
+        (signed)(((xip_currtime() - (x)) * USEC_PER_SEC) / LATCH)
+
+#endif				/* __ARCH_MXC_MTD_XIP_H__ */
